tamr_unify_client/dataset/uri.py

```diff
@@ -0,0 +1,45 @@
+class DatasetURI:
+    """
+    Upstream dataset being used by dataset being queried.
+
+    See https://docs.tamr.com/v2019.013/reference#retrieve-upstream-datasets
+
+    :param client: Queried dataset's client.
+    :type client: :class:`~tamr_unify_client.client.Client`
+    :param uri: Queried datset's dataset ID.
+    :type uri: :py:class:`str`
+    """
+
+    def __init__(self, client, uri):
+        self.client = client
+        self._uri = uri
+
+    @property
+    def resource_id(self):
+        """:type: str"""
+        return self._uri.split("/")[-1]
+
+    @property
+    def relative_id(self):
+        """:type: str"""
+        return "datasets/" + self.resource_id
+
+    @property
+    def uri(self):
+        """:type: str"""
+        return self._uri
+
+    def dataset(self):
+        """Dataset that queried dataset is pulling data from.
```

Comment:
I don't know if this exactly represents what this is, it's the dataset this URI identifies